ob1k-cache/src/main/java/com/outbrain/ob1k/cache/memcache/compression/CompressionAlgorithm.java

```diff
@@ -0,0 +1,72 @@
+package com.outbrain.ob1k.cache.memcache.compression;
+
+import com.github.luben.zstd.Zstd;
+import net.jpountz.lz4.LZ4Factory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.function.Function;
+import java.util.zip.DataFormatException;
+import java.util.zip.Deflater;
+import java.util.zip.Inflater;
+
+/**
+ * Created by bshushi on 27/03/2017.
+ */
+public enum CompressionAlgorithm {
+  LZ4,
+  ZSTANDARD,
+  GZIP;
+
+  private static LZ4Factory lz4Factory = LZ4Factory.fastestInstance();
+
+  public Compressor getInstance() {
+    switch (this) {
```

Comment:
no need for the switch
just make #getInstance be abstract method and implement each one at the specific enum value

for exmaple:

LZ4 {
  ...getInstance() {
      return new ...();
  }
}